File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length, maxProfit = 0;
        int left = 0, right = 1;
        while(right<n){
            if(prices[left] < prices[right]){
                maxProfit = Math.max(maxProfit, prices[right] - prices[left]);
            }else{
                left = right;
            }
            right++;
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        right = 0
        result, current_min = -sys.maxsize, sys.maxsize
        while right < len(prices):
            if prices[right] < current_min:
                current_min = prices[right]
            
            result = max(result, prices[right] - current_min)
            right += 1
        
        return result
        